src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -2919,6 +2991,74 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
                 break;
             }
 
+            size_t elementSize  = genTypeSize(simdBaseType);
```

Comment:
This logic needs to be fixed in `IsValidForShuffle` now instead, it was refactored a few days ago to allow handling cases that weren't constant in importation.

Reply:
Merged correctly